src/test/java/jenkins/plugins/logstash/LogstashWriterTest.java

```diff
@@ -0,0 +1,304 @@
+  }
+
+  @Test
+  public void writeSuccessConnectionBroken() throws Exception {
+    Mockito.doNothing().doThrow(new IOException("BOOM!")).doNothing().when(mockDao).push(anyString());
+    LogstashWriter los = createLogstashWriter(mockBuild, errorBuffer, "http://my-jenkins-url", mockDao, mockBuildData);
+
+
+    String msg = "test";
+    String exMessage = "[logstash-plugin]: Failed to send log data to REDIS:localhost:8080.\n" +
+      "[logstash-plugin]: No Further logs will be sent to localhost:8080.\n" +
+      "java.io.IOException: BOOM!";
+
+    errorBuffer.reset();
+
+    // Unit under test
+    los.write(msg);
+
+    // Verify results
+    assertEquals("Results don't match", "", errorBuffer.toString());
+
+    // Break the dao connnection
+    errorBuffer.reset();
+
+    // Unit under test
+    los.write(msg);
+
+    // Verify results
+    assertTrue("Results don't match", errorBuffer.toString().startsWith(exMessage));
+    assertTrue("Connection not broken", los.isConnectionBroken());
+
+    // Verify logs still write but on further calls are made to dao
+    errorBuffer.reset();
+    // Unit under test
+    los.write(msg);
+
+    // Verify results
+    assertEquals("Results don't match", "", errorBuffer.toString());
+
+    //Verify calls were made to the dao logging twice, not three times.
+    verify(mockDao, times(2)).buildPayload(Matchers.eq(mockBuildData), Matchers.eq("http://my-jenkins-url"), Matchers.anyListOf(String.class));
+    verify(mockDao, times(2)).push("{\"data\":{},\"message\":[\"test\"],\"source\":\"jenkins\",\"source_host\":\"http://my-jenkins-url\",\"@version\":1}");
+    verify(mockDao).getIndexerType();
+    verify(mockDao, times(2)).getDescription();
+  }
+
+  @Test
+  public void writeBuildLogGetLogError() throws Exception {
+    // Initialize mocks
+    when(mockBuild.getLog(3)).thenThrow(new IOException("Unable to read log file"));
+
+    LogstashWriter writer = createLogstashWriter(mockBuild, errorBuffer, "http://my-jenkins-url", mockDao, mockBuildData);
+    assertEquals("Errors were written", "", errorBuffer.toString());
+
+    // Unit under test
+    writer.writeBuildLog(3);
+
+    // Verify results
+    verify(mockBuild).getLog(3);
+
+    List<String> expectedErrorLines =  Arrays.asList(
+      "[logstash-plugin]: Unable to serialize log data.",
+      "java.io.IOException: Unable to read log file");
+    verify(mockDao).push("{\"data\":{},\"message\":[\"test\"],\"source\":\"jenkins\",\"source_host\":\"http://my-jenkins-url\",\"@version\":1}");
+    verify(mockDao).buildPayload(eq(mockBuildData), eq("http://my-jenkins-url"), logLinesCaptor.capture());
+    List<String> actualLogLines = logLinesCaptor.getValue();
+
+    assertEquals("The exception was not sent to Logstash", actualLogLines.get(0), expectedErrorLines.get(0));
+    assertEquals("The exception was not sent to Logstash", actualLogLines.get(1), expectedErrorLines.get(1));
```

Comment:
Instead of looping for each line, I just specified that the first two lines should match.